specifyweb/frontend/js_src/lib/components/WorkBench/WbUtils.tsx

```diff
@@ -6,35 +6,232 @@
  *
  */
 
-import type Handsontable from 'handsontable';
 import React from 'react';
 import _ from 'underscore';
 
 import { wbText } from '../../localization/workbench';
 import { f } from '../../utils/functools';
 import type { RA, WritableArray } from '../../utils/types';
-import { filterArray } from '../../utils/types';
-import { camelToKebab, clamp } from '../../utils/utils';
-import { Backbone } from '../DataModel/backbone';
-import { softFail } from '../Errors/Crash';
-import { LeafletMap } from '../Leaflet/Map';
-import { getLocalitiesDataFromSpreadsheet } from '../Leaflet/wbLocalityDataExtractor';
+import { camelToKebab } from '../../utils/utils';
 import { hasPermission } from '../Permissions/helpers';
 import type { WbSearchPreferences } from './AdvancedSearch';
 import {
   getInitialSearchPreferences,
   WbAdvancedSearch,
 } from './AdvancedSearch';
 import type { WbCellCounts } from './CellMeta';
-import { CoordinateConverter } from './CoordinateConverter';
-import { getSelectedLocalities, WbGeoLocate } from './GeoLocate';
 import { getHotPlugin } from './handsontable';
-import { getSelectedLast, getVisualHeaders } from './hotHelpers';
-import type { WbView } from './WbView';
+import { getSelectedLast } from './hotHelpers';
+import type { Workbench } from './WbView';
+import { Input } from '../Atoms/Form';
+import { LocalizedString } from 'typesafe-i18n';
+import { Button } from '../Atoms/Button';
+import { className } from '../Atoms/className';
+import { commonText } from '../../localization/common';
+import { useBooleanState } from '../../hooks/useBooleanState';
+import { icons } from '../Atoms/Icons';
+
+function Navigation({
+  name,
+  label,
+  totalCount,
+  utils,
+  spreadsheetContainer,
+}: {
+  readonly name: keyof WbCellCounts;
+  readonly label: LocalizedString;
+  readonly totalCount: number;
+  readonly utils: WbUtils;
+  readonly spreadsheetContainer: any;
```

Comment:
don't use any!

it looks like you are using `spreadsheetContainer` only to toggle a class name - something that can be refactored to use React instead

and if you still need `spreadsheetContainer` after that, it's type should be `React.Ref<HTMLElement>`